jni/jni.cabal

```diff
@@ -48,9 +46,16 @@ library
     constraints >=0.8,
     deepseq >=1.4.2,
     inline-c >=0.6
-  if flag(ghc-head)
+  if flag(linear-types)
+    hs-source-dirs: src/linear-types
+    build-depends:
+      linear-base ==0.1.0.0
     exposed-modules:
       Data.Singletons
+      Foreign.JNI.Linear
+      Foreign.JNI.Types.Linear
+      Linear.Extras
```

Comment:
These additions to `linear-base` should be upstreamed. And meanwhile should not be exported to the user, since the module name will sure clash.